### PR TITLE
Beautify the display of large amounts on small devices

### PR DESCRIPTION
**Pull request by EffectDoplera**
Screenshot 2023-12-21 at 14 54 13

### CODE REVIEW

**Comment by astsu-dev** [comment]
Hi @EffectDoplera, thanks for your contribution! I really appreciate it, and congratulations on being the first contributor! I like your idea, and I'll review your pull request when I have some free time.

File: src/entities/currency/lib.ts [hunk 1]
```diff
@@ -24,7 +24,9 @@ export const createCurrencyAmountString = ({
   if (symbolPosition === "right") {
     parts.reverse();
   }
-  const currencyString = parts.join(hasSpaceBetweenAmountAndSymbol ? " " : "");
+  const currencyString = parts
+    .join(hasSpaceBetweenAmountAndSymbol ? " " : "")
+    .replace(/\B(?=(\d{3})+(?!\d))/g, ",");
```
Comment (astsu-dev) [inline src/entities/currency/lib.ts]: Move this regex to a constant

Comment (astsu-dev) [reply src/entities/currency/lib.ts]: To improve this, we can add a checkbox to the currency page for grouping numbers. Users can choose whether to group numbers or not. Additionally, please create unit tests for this regex, and make sure to test decimal numbers.

**Comment by astsu-dev** [comment]
Hi @EffectDoplera. Could you share your screen resolution and PPI with me? By the way, are you aware that these cards are scrollable? Regardless, I believe it might be a good idea to display these cards vertically for smaller screens, but let's talk more about the option of scrollable cards.

**Comment by EffectDoplera** [comment]
> Hi @EffectDoplera. Could you share your screen resolution and PPI with me? By the way, are you aware that these cards are scrollable? Regardless, I believe it might be a good idea to display these cards vertically for smaller screens, but let's talk more about the option of scrollable cards.

Hi @astsu-dev. Of course, the specs of the device I'm using are 1080 x 2400 pixels, 20:9 ratio (~398 ppi density). I've seen that cards can scroll, but it's not always convenient for currencies that have a large number of zeros.

File: src/shared/lib/regex.test.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { it, describe, expect } from "vitest";
+
+import { numbersGroupingRegex } from "./regex";
+
+describe("numbersGroupingRegex", () => {
+  it("int numbers", () => {
```
Comment (astsu-dev) [inline src/shared/lib/regex.test.ts]: It's better to define a group of tests with `describe` function and don't have `it` inside `it`.

File: src/shared/lib/regex.test.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    it("should match numbers with thousands separators", () => {
+      expect("123456789.01".replace(numbersGroupingRegex, ",")).toBe(
+        "123,456,789.01",
+      );
+    });
+
+    it("should not match numbers less than four digits", () => {
+      expect("123.01".replace(numbersGroupingRegex, ",")).toBe("123.01");
+    });
+  });
```
Comment (astsu-dev) [inline src/shared/lib/regex.test.ts]: Add the test case for numbers with more than 3 digits after the decimal point. It is the most important test case for this regex. It should not group digits after the decimal point. It's useful when you use currencies with the high precision. I often use crypto currencies, where number of digits after the decimal point can be up to 18. For example, Bitcoin decimal representation usually has 8 digits after the decimal point.

**Pull request closed without merging**